lib/options-manager.js

```diff
@@ -243,8 +244,30 @@ const buildConfig = options => {
 		Object.assign(config.rules, options.rules);
 	}
 
-	if (options.settings) {
-		config.baseConfig.settings = options.settings;
+	config.baseConfig.settings = Object.assign({}, options.settings);
+	let importResolver = config.baseConfig.settings['import/resolver'] || {};
+
+	// Convert import/resolver: 'x' => import/resolver: {x: {}}
+	if (typeof importResolver === 'string') {
+		const resolverName = importResolver;
+		importResolver = {};
+		importResolver[resolverName] = {};
+	}
+
+	if (options.webpack) {
+		importResolver.webpack = options.webpack === true ? {} : options.webpack;
+	} else if (!importResolver.webpack) {
+		// If a webpack config file exists, add the import resolver automatically
+		const webpackConfigPath = findUp.sync('webpack.config.js', {cwd: options.cwd});
+		if (webpackConfigPath) {
+			importResolver.webpack = {config: webpackConfigPath};
+		}
+	}
+
+	if (Object.keys(importResolver).length > 0) {
+		config.baseConfig.settings['import/resolver'] = importResolver;
+	} else {
+		delete config.baseConfig.settings['import/resolver'];
 	}
```

Comment:
Would be nice if the logic for the Webpack stuff was moved out into its own function. The function here is already very large.